test/Elastic.Apm.AspNetCore.Tests/AspNetCoreBasicTests.cs

```diff
@@ -244,8 +244,8 @@ public async Task HomeSimplePagePostTransactionTest(bool withDiagnosticSourceOnl
 
 			_agent.Service.Framework.Name.Should().Be("ASP.NET Core");
 
-			var aspNetCoreVersion = Assembly.Load("Microsoft.AspNetCore").GetName().Version.ToString();
-			_agent.Service.Framework.Version.Should().Be(aspNetCoreVersion);
+			var aspNetCoreMajorMinorVersion = Assembly.Load("Microsoft.AspNetCore").GetName().Version.ToString(2);
```

Comment:
Why do we change this? 🤔  We set `Service.Framework.Version` to the assembly version (without truncating it). I'd assert on the whole value still.

Reply:
This test consistently fails for me locally on 2.1 and 3.1 with diagnostic source `true`. Debugging the issue, it's related to

https://github.com/elastic/apm-agent-dotnet/blob/dbad2d54ffd2bc7ba7a27e96506f7f3595765fd6/src/Elastic.Apm.Extensions.Hosting/HostBuilderExtensions.cs#L84-L98

With diagnostic source `true`, `Microsoft.AspNetCore` assembly is not found, so resorts to assemblies containing `Microsoft.AspNetCore`, which ends up picking up `Microsoft.AspNetCore.Mvc.Core` that has a different patch version for 2.1 and 3.1 (for 2.1,  `Microsoft.AspNetCore` is 2.1.7.0 and  `Microsoft.AspNetCore.Mvc.Core` is 2.1.16.0)

![Assembly Version](https://user-images.githubusercontent.com/208231/99738208-ac4d8080-2b15-11eb-91fc-df1a8ed3bfc7.png)

Be curious to see which assembly your local test picks for this.

Reply:
```
> dotnet --list-sdks

2.1.300 [C:\Program Files\dotnet\sdk]
2.1.801 [C:\Program Files\dotnet\sdk]
2.2.103 [C:\Program Files\dotnet\sdk]
2.2.104 [C:\Program Files\dotnet\sdk]
2.2.401 [C:\Program Files\dotnet\sdk]
3.0.100 [C:\Program Files\dotnet\sdk]
3.1.100 [C:\Program Files\dotnet\sdk]
3.1.200 [C:\Program Files\dotnet\sdk]
5.0.100-preview.8.20362.3 [C:\Program Files\dotnet\sdk]
5.0.100 [C:\Program Files\dotnet\sdk]
```

Reply:
Hahh, I have the same problem on `2.1`.

But this would mean we may not report the correct patch version. By looking at this, there is a complete chaos on `2.1`.

I have:
```
dotnet --list-sdks
2.1.612 [C:\Program Files\dotnet\sdk]
... and more 2.2, etc...
```

Then, when I run an ASP.NET Core app, assemblies are loaded from the folder `2.1.23`: 
```
Program Files\dotnet\shared\Microsoft.AspNetCore.App\2.1.23
```

With the following assemblies and versions in it:
```
Microsoft.AspNetCore.Mvc.Core, Version=2.1.16.0,
Microsoft.AspNetCore, Version=2.1.7.0
```

These are 4 different versions and at this point I'm not even sure what we should report.

If we ignore the patch part of the version, it could be that we actually ignore a bug. This definitely should not block this PR, on the other this may be a real bug here (probably a small one, but still...)

With this we'll report `Framework{Name: ASP.NET Core, Version: 2.1.16.0}`, both on your machine and on mine, while you have sdk `2.1.801` and I have sdk `2.1.612`.

Reply:
Ok, I'll revert this change so that this PR can be merged, and I'll open a new issue to discuss

Reply:
Opened https://github.com/elastic/apm-agent-dotnet/issues/1061